### DISCUSSION TITLE
Nivo/Bar Customize label, add two labels

### DISCUSSION THREAD

**Discussion by Carlfredo-dev** [plouc/nivo#2660] [General]
Hello!

I am using the nivo/bar component in a project made with react. Specifically the "ResponsiveBar" component and I need to add another value or additional field below it in the value label, and for both to be centered both vertically and horizontally.

I have something like this but I can't get what I requested.

Code:
           <ResponsiveBar
              data={data}
              theme={theme}
              keys={["velocidad"]}
              indexBy="conductor"
              margin={{ top: 30, right: 10, bottom: 50, left: 80 }}
              padding={0.2}
              valueFormat={(value) => FormatNumber(value)}
              colors={["#DD6D53"]}
              borderColor={{
                from: "color",
                modifiers: [["darker", "1.6"]],
              }}
              axisTop={null}
              axisRight={null}
              axisBottom={{
                tickSize: 5,
                legend: "Conductor",
                legendPosition: "middle",
                legendOffset: 40,
                format: (value) => FormatNumber(value),
              }}
              axisLeft={{
                tickSize: 5,
                tickPadding: 5,
                tickRotation: 0,
                legend: "Velocidad",
                legendPosition: "middle",
                legendOffset: -60,
                format: (value) => FormatNumber(value),
              }}
              enableTotals={true}
              totalsOffset={11}
              labelSkipWidth={12}
              labelSkipHeight={12}
              labelPosition="middle"
              label={(bar) => (
                <tspan>
                  <tspan textAnchor="middle">{bar.data.velocidad}</tspan>
                  <tspan textAnchor="middle">
                    {bar.data.conductor}
                  </tspan>
                </tspan>
              )}
              labelTextColor="#ffffff"
              legends={[]}
              isInteractive={true}
              role="application"
              ariaLabel="Gráfica Patentes con más Km."
              // }
            />
            
            
![image](https://github.com/user-attachments/assets/58708ab9-274f-4e72-92b2-b9df66dd263a)

This currently looks like this, I need the name to be with a line break, how could I achieve it?

ps: Sorry my English, I am Spanish speaking, thank you very much in advance for the help.

**Comment by manimovassagh** [comment]
The `label` prop on `ResponsiveBar` expects either a string (a data key) or a function that returns an SVG element. The issue with your `<tspan>` approach is that you need `dy` attributes to create the line break, and the outer `<tspan>` needs the inner ones positioned correctly.

Here is how to render two centered lines inside each bar:

```jsx
label={(bar) => (
  <tspan>
    <tspan x="0" dy="-0.6em" textAnchor="middle">
      {bar.data.velocidad}
    </tspan>
    <tspan x="0" dy="1.2em" textAnchor="middle">
      {bar.data.conductor}
    </tspan>
  </tspan>
)}
```

The `x="0"` resets the horizontal position for each line, and `dy` offsets vertically. `-0.6em` pushes the first line up from center, `1.2em` puts the second one below.

If this still does not center properly (which can happen depending on the nivo version), use a custom layer instead for full control:

```jsx
const CustomBarLabel = ({ bars }) => (
  <g>
    {bars.map((bar) => (
      <text
        key={bar.key}
        x={bar.x + bar.width / 2}
        y={bar.y + bar.height / 2}
        textAnchor="middle"
        dominantBaseline="central"
        fill="#ffffff"
        fontSize={12}
      >
        <tspan x={bar.x + bar.width / 2} dy="-0.6em">
          {bar.data.data.velocidad}
        </tspan>
        <tspan x={bar.x + bar.width / 2} dy="1.2em">
          {bar.data.data.conductor}
        </tspan>
      </text>
    ))}
  </g>
);

<ResponsiveBar
  // ... your other props
  enableLabel={false}
  layers={["grid", "axes", "bars", CustomBarLabel, "markers", "legends"]}
/>
```

The custom layer approach is more reliable for multi-line labels since you have full SVG positioning control.
